File: Bench.hs
{-# LANGUAGE Strict #-}

import Data.List (foldl')

data Rectangle = Rectangle { lx :: Float
                           , by :: Float
                           , rx :: Float
                           , ty :: Float } deriving Show

r = Rectangle

union (Rectangle alx aby arx aty)
      (Rectangle blx bby brx bty) =
  Rectangle (min alx blx)
            (min aby bby)
            (max arx brx)
            (max aty bty)

-- runs for minutes, before being manually killed
-- no one knows how to get folds over very large
-- lists to work in haskell >.>

main :: IO ()
main = putStrLn
     $ show
     $ foldl' union (r 25 25 200 200)
     $ map (r 20 0 100) [0..1e9]

File: bench.cpp
#include <algorithm>
#include <cstdio>

struct Rectangle {
    double lx, by, rx, ty;
};

Rectangle Union(const Rectangle& r1, const Rectangle& r2) {
    return Rectangle{ std::min(r1.lx, r2.lx)
                    , std::min(r1.by, r2.by)
                    , std::max(r1.rx, r2.rx)
                    , std::max(r1.ty, r2.ty) };
}

int main() {
    Rectangle k{ 25, 25, 200, 200 };
    for (int i = 0; i < 1000000000; ++i)
      k = Union(k, Rectangle{ 20, 0, 100, (double)i });

    printf("Rectangle lx:%f by:%f rx:%f ty:%f\n", k.lx, k.by, k.rx, k.ty);
    return 0;
}

